### ISSUE TITLE
Add `xsimlab.create_setup` function

### ISSUE THREAD

**Issue by benbovy** [benbovy/xarray-simlab#2]
Currently, to create a simulation setup we do something like this:

```python
import xarray as xr
import xsimlab

in_ds = xr.Dataset()   

in_ds.xsimlab.use_model(fastscape_base_model) 

in_ds.xsimlab.set_input_vars('grid', x_size=401, y_size=201, x_length=4e5, y_length=2e5)
in_ds.xsimlab.set_input_vars('topography', elevation=('node', np.random.rand(401 * 201)))
in_ds.xsimlab.set_input_vars('spower', k_coef=7e-5, m_exp=0.4, n_exp=1)
in_ds.xsimlab.set_input_vars('uplift', u_coef=2e-3)

in_ds.xsimlab.set_master_clock('time', end=1.5e6, step=1e3)

in_ds.xsimlab.set_snapshot_clock('out', end=1.5e6, step=1e4)
in_ds.xsimlab.set_snapshot_vars('out', topography='elevation')

in_ds.xsimlab.set_snapshot_clock('out2', end=1.5e6, step=1e5)
in_ds.xsimlab.set_snapshot_vars('out2', area='area')

in_ds.xsimlab.set_snapshot_vars(None, grid=['x', 'y'])
```

Alternatively we could do:

```python
in_ds = xsimlab.create_setup(
    fastscape_base_model,
    input_vars={
        'grid': {'x_size': 401, 'y_size': 201, 'x_length': 4e5, 'y_length': 2e5},
        'topography': {'elevation': ('node', np.random.rand(401 * 201))},
        'spower': {'k_coef': 7e-5, 'm_exp': 0.4, 'n_exp': 1},
        'uplift': {'u_coef': 2e-3},
    },
    clocks={
        'time': {'end': 1.5e6, 'step': 1e3},
        'out': {'end': 1.5e6, 'step': 1e4},
        'out2': {'end': 1.5e6, 'step': 1e5},
    },
    master_clock='time',
    snapshot_vars={
        'out': {'topography': 'elevation'},
        'out2': {'area': 'area'},
        None: {'grid': ('x', 'y')},
    }
)
```

The latter has the advantage to use a more declarative syntax and avoid using multiple in-place operations.

It is often useful to take an existing setup and change only a few values. For this case, we could add a `use_dataset` keyword argument to `xsimlab.setup` (that defaults to `None`), or maybe even better add a `Dataset.xsimlab.update_setup()` method that has the same signature than `xsimlab.create_setup`.

**Comment by benbovy** [comment]
Some other useful functions or methods that would form togheter a consistent API:

- `xsimlab.load_setup(model, filename)`: loads a setup given in an input file (e.g., yaml format) and returns a new Dataset
- `Dataset.xsimlab.clean_setup(model)`: drop in Dataset all variables that are not inputs of `model`

A common use case: we create a slightly different version of a model and we want to re-use an existing setup. This would look like:

```python
alt_model = model.update_processes({...})

in_ds = (
    xsimlab.load_setup(model, 'input.yml')
    .clean_setup(alt_model)
    .update_setup(alt_model, input_vars={...})
)
```

**Comment by benbovy** [comment]
Fixed in #9.
